services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -15,514 +15,415 @@
-        }
-        return attributeValue.ss().stream().collect(Collectors.toSet());
+        return get(attributeName, EnhancedType.setOf(String.class));
+
     }
 
     @Override
     public Set<SdkNumber> getNumberSet(String attributeName) {
-        AttributeValue attributeValue = attributeValueMap.get(attributeName);
-        if (attributeValue == null || !attributeValue.hasNs()) {
-            return null;
-        }
-        return attributeValue.ns().stream().map(SdkNumber::fromString).collect(Collectors.toSet());
+        return get(attributeName, EnhancedType.setOf(SdkNumber.class));
     }
 
     @Override
-    public Set<SdkBytes> getSdkBytesSet(String attributeName) {
-        AttributeValue attributeValue = attributeValueMap.get(attributeName);
-        if (attributeValue == null || !attributeValue.hasBs()) {
-            return null;
-        }
-        return attributeValue.bs().stream()
-                             .map(item -> SdkBytes.fromByteArray(item.asByteArrayUnsafe()))
-                             .collect(Collectors.toSet());
+    public Set<SdkBytes> getBytesSet(String attributeName) {
+        return get(attributeName, EnhancedType.setOf(SdkBytes.class));
     }
 
     @Override
     public <T> List<T> getList(String attributeName, EnhancedType<T> type) {
-
-        AttributeConverter<T> attributeConverter = attributeConverterProviders.converterFor(type);
-        if (attributeConverter == null) {
-            throw new IllegalArgumentException("type " + type + " is not found in AttributeConverterProviders");
-        }
-
-        AttributeValue attributeValue = attributeValueMap.get(attributeName);
-        if (attributeValue == null || !attributeValue.hasL()) {
-            return null;
-        }
-        return attributeValue.l().stream().map(
-            value -> attributeConverterProviders.converterFor(type).transformTo(value)).collect(Collectors.toList());
+        return get(attributeName, EnhancedType.listOf(type));
     }
 
     @Override
-    public List<?> getList(String attributeName) {
-        AttributeValue attributeValue = attributeValueMap.get(attributeName);
-        if (attributeValue == null || !attributeValue.hasL()) {
-            return null;
-        }
-        return toSimpleList(attributeValue.l());
+    public <K, V> Map<K, V> getMap(String attributeName, EnhancedType<K> keyType, EnhancedType<V> valueType) {
+        return get(attributeName, EnhancedType.mapOf(keyType, valueType));
     }
 
     @Override
-    public <T> Map<String, T> getMap(String attributeName, EnhancedType<T> type) {
-        validateConverter(type);
-        AttributeValue attributeValue = attributeValueMap.get(attributeName);
-        if (attributeValue == null || !attributeValue.hasM()) {
+    public String getJson(String attributeName) {
+        AttributeValue attributeValue = attributeValueMap.getValue().get(attributeName);
+        if (attributeValue == null) {
             return null;
```

Comment:
Technically a null value in JSON is `null` so shouldn't this return `"null"`?

Reply:
Agree, but that would mess up with the application code which assumes it to be as valid json that would be null (nothing at all) , {} (empty) , or something in {}
Also V1 we return null https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/document/Item.java#L925-L929